tests/multi_ptr/multi_ptr_access_members.h

```diff
@@ -0,0 +1,203 @@
+                                                      sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_result_buf(&get_result, sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_raw_result_buf(&get_raw_result,
+                                               sycl::range<1>(1));
+      sycl::buffer<bool, 1> get_decorated_result_buf(&get_decorated_result,
+                                                     sycl::range<1>(1));
+
+      queue.submit([&](sycl::handler &cgh) {
+        auto dereference_value_acc =
+            dereference_value_buffer
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto dereference_op_value_acc =
+            dereference_op_value_buffer
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto get_member_value_acc =
+            get_member_value_buffer
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto get_raw_member_value_acc =
+            get_raw_member_value_buffer
+                .template get_access<sycl::access_mode::write>(cgh);
+
+        auto dereference_result_acc =
+            dereference_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto dereference_op_result_acc =
+            dereference_op_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+        auto get_result_acc =
+            get_result_buf.template get_access<sycl::access_mode::write>(cgh);
+        auto get_raw_result_acc =
+            get_raw_result_buf.template get_access<sycl::access_mode::write>(
+                cgh);
+        auto get_decorated_result_acc =
+            get_decorated_result_buf
+                .template get_access<sycl::access_mode::write>(cgh);
+
+        auto acc_for_multi_ptr =
+            val_buffer.template get_access<sycl::access_mode::read>(cgh);
+        cgh.single_task([=] {
+          const multi_ptr_t multi_ptr(acc_for_multi_ptr);
+
+          // Dereference and multi_ptr::operator->() available only when:
+          // !std::is_void<sycl::multi_ptr::value_type>::value
+          if constexpr (!std::is_void_v<multi_ptr_t::value_type>) {
+            // Check dereference operator return value and type correctness
+            dereference_result_acc[0] =
+                std::is_same_v<decltype(*multi_ptr), multi_ptr_t::reference>;
+            dereference_value_acc[0] = *multi_ptr;
+            // Check operator->() return value and type correctness
+            dereference_op_result_acc[0] =
+                std::is_same_v<decltype(multi_ptr.operator->()),
+                               multi_ptr_t::pointer>;
+            dereference_op_value_acc[0] = *(multi_ptr.operator->());
+          }
+
+          // Check get() return value and type correctness
+          get_result_acc[0] =
+              std::is_same_v<decltype(multi_ptr.get()), multi_ptr_t::pointer>;
+          get_member_value_acc[0] = *(multi_ptr.get());
```

Comment:
Maybe we shouldn't check this value if get() returns decorated pointer since we can't be sure that it can be deferenced like that&